### ISSUE TITLE
Hide American express icon in credit card button for some currencies

### ISSUE THREAD

**Issue by ScottDowne** [MozillaFoundation/Advocacy#613]
Detailed description of work: Some Stripe currencies don't support American Express. We should hide the icon for these currencies.

<< Matt H. removed the due date since this is "To Be Prioritized" >>

**Comment by Matt-Holland** [comment]
@roseregina can you check what is the user experience when someone tries to donate via Stripe, using American Express, in a currency which is not supported? What sort of error message do they get?

**Comment by ScottDowne** [comment]
We can do test American Express donations using this card number 378282246310005 on donate.mofostaging.net

Right now, it's silently fails, and the user is left on the donate page wondering if it worked or not.

I think this is a great find, and we need to fix it.

**Comment by ScottDowne** [comment]
More test cards can be found here: https://stripe.com/docs/testing#cards

**Comment by ScottDowne** [comment]
Unfortunatly, Stripe looks to just give a "card_declined" error, and not anything more specific. We can display what we know, and this is simply that the card was declined.

We can also not display the amex logo for these currencies.

What we can not do is display an error along the lines of "Sorry American Express is not supported, try another currency or another card."

**Comment by ScottDowne** [comment]
We can maybe display a guess at the error. And say something like "If you're using American Express with currency x, that won't work, or something."

**Comment by roseregina** [comment]
Does discover work? it looks like it did when I very quickly checked this morning, that that is the only kind of card I have gotten emails out.

**Comment by ScottDowne** [comment]
@roseregina Glad you pointed this out. Looks like Discover is ONLY supported by USD.

I created a pull request to better expose these error cases we're finding, so once it lands, we'll have a better idea who is hitting this, and who is hitting more unknown errors. To be clear, this error message isn't going to allow them to donate, and probably won't reduce frustration, it'll just help us debug to go forward and possibly see bugs we can fix.

**Comment by ScottDowne** [comment]
It's also good info to know so we can better respond to complaints, or maybe add to our FAQ.

**Comment by roseregina** [comment]
Is there any way of setting up an email alert for events that end silent fails? There may be far too many to do anything about, but I wonder about using desk to send out emails to people who would otherwise think that they had donated to us.

**Comment by ScottDowne** [comment]
I think ideally we fix it so it doesn't silently fail. And they get an
error message on the form instead of later in an email.

**Comment by roseregina** [comment]
I meant an email to one of us to figure out what is going on. 

though, interestingly, there have been a few people who in spite of (I assume) getting some kind of error message from a bank decline, think they have donated and are wondering why they haven't gotten a thank you email. lesson from that might just be that people don't always pay much attention.

**Comment by ScottDowne** [comment]
This got done during Hawaii.
